db/migrate/20170718111834_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,174 @@
+        join_sql = definitions.project(definitions[column.to_sym])
+                              .where(definitions[:id].eq(containers[:container_definition_id])).to_sql
+        Container.update_all("#{column} = (#{join_sql})")
+      end
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      port_configs = Arel::Table.new(:container_port_configs)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(port_configs[:container_definition_id])).to_sql
+      ContainerPortConfig.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch container_definition_id with container_id for container_port_configs") do
+      env_vars = Arel::Table.new(:container_env_vars)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(env_vars[:container_definition_id])).to_sql
+      ContainerEnvVar.update_all("container_definition_id = (#{join_sql})")
+    end
+
+    say_with_time("switch resource_id with container_id, resource_type to 'Container' for security_contexts") do
+      security_contexts = Arel::Table.new(:security_contexts)
+      join_sql = containers.project(containers[:id])
+                           .where(containers[:container_definition_id].eq(security_contexts[:resource_id])
+                           .and(security_contexts[:resource_type].eq(Arel::Nodes::Quoted.new('ContainerDefinition')))).to_sql
+      SecurityContext.where(:resource_type => 'ContainerDefinition').update_all("resource_type = 'Container', resource_id = (#{join_sql})")
+    end
+
+    MiqQueue.where(:method_name => "purge_timer", :class_name => 'ContainerDefinition').destroy_all
+
+    # relationships
+    rename_column :container_port_configs, :container_definition_id, :container_id
+    rename_column :container_env_vars, :container_definition_id, :container_id
+
+    remove_column :containers, :container_definition_id
+    drop_table :container_definitions
+  end
+
+  def down
+    create_table :container_definitions do |t|
+      t.belongs_to :ems, :type => :bigint       # reconstructed columns
+      t.string     :ems_ref
+      t.bigint     :old_ems_id
+      t.timestamp  :deleted_on
+      t.string     :name
+      t.string     :image                       # copied over columns
+      t.string     :image_pull_policy
+      t.string     :memory
+      t.float      :cpu_cores
+      t.belongs_to :container_group, :type => :bigint
+      t.boolean    :privileged
+      t.bigint     :run_as_user
+      t.boolean    :run_as_non_root
+      t.string     :capabilities_add
+      t.string     :capabilities_drop
+      t.text       :command
+      t.bigint     :container_id                # temp column
+    end
+
+    add_index :container_definitions, :deleted_on
```

Comment:
You probably want to add the index *after* you've done the data migration, otherwise each insert needs to update the index individually making it slower overall.